File: src/commands/CommitCommand.ts
import { InvalidCommandError } from "../common/helpers/errors/minigit.js";
import logger from "../common/helpers/logger.js";
import { commit } from "../core/commit/commit.js";
import { Command } from "./types.js";

export class CommitCommand implements Command {
  execute(args: string[]): void {
    const [flag, message] = args;

    if (!flag || flag !== "-m" || !message) {
      throw new InvalidCommandError(
        "please provide -m flag along with commit message to commit your changes",
      );
    }

    const commitId = commit(message);
    logger.logWithNewLine(`Committed changes with commit Id: ${commitId}`);
  }
}

File: src/cli/commands/commit_command.cpp
#include <print>

#include "minigit/cli/commands/commit_command.hpp"
#include "minigit/common/util/error.hpp"

namespace minigit::cli {

    int CommitCommand::execute(repo::Repository* repo, std::span<const std::string_view> args) const
    {
        if (args.size() < 2 || args[0] != "-m")
        {
            throw Error(
                ErrorCode::InvalidCommand,
                "INVALID_COMMAND_ERROR: please provide -m flag along with commit message to "
                "commit your changes");
        }

        const auto commit_id = repo->commit(std::string(args[1]));
        std::println("Committed changes with commit Id: ", commit_id.to_string());
        return 0;
    }

} // namespace minigit::cli
